server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -357,6 +359,22 @@ public void applyClusterState(ClusterChangedEvent event) {
         }
     }
 
+    /**
+     * Gets the {@link RepositoryData} for the given repository.
+     *
+     * @param repositoryName repository name
+     * @param listener       listener to pass {@link RepositoryData} to
+     */
+    public void getRepositoryData(final String repositoryName, final ActionListener<RepositoryData> listener) {
+        try {
+            Repository repository = repository(repositoryName);
+            assert repository != null; // should only be called once we've validated the repository exists
```

Comment:
Couldn't the repository have been removed after we checked the repo exists? I think we should throw a RepositoryMissingException here instead of NullPointerException

Reply:
In `repository(repositoryName)` we do actually throw `RepositoryMissingException`. The assertion is a little pointless but I don't think we're ever throwing an NPE here